src/linear_algebra/projective_space/basic.lean

```diff
@@ -0,0 +1,218 @@
+as well as the bijection between said projectivization and the collection of all one
+dimensional subspaces of the vector space.
+
+## Constructing terms of `projectivization K V`.
+We have three ways to construct terms of `projectivization K V`:
+- `projectivization.mk K v hv` where `v : V` and `hv : v ≠ 0`.
+- `projectivization.mk' K v` where `v : { w : V // w ≠ 0 }`.
+- `projectivization.mk'' H h` where `H : submodule K V` and `h : finrank H = 1`.
+
+## Other definitions
+- For `v : projectivization K V`, `v.submodule` gives the corresponding submodule of `V`.
+- `projectivization.equiv_submodule` is the equivalence between `projectivization K V`
+  and `{ H : submodule K V // finrank H = 1 }`.
+- For `v : projectivization K V`, `v.rep : V` is a representative of `v`.
+
+## Projects
+Everything in this file can be done for `division_ring`s instead of `field`s, but
+this would require a significant refactor of the results from
+`linear_algebra.finite_dimensional` and its imports.
+
+-/
+
+variables (K V : Type*) [field K] [add_comm_group V] [module K V]
+
+/-- The setoid whose quotient is the projectivization of `V`. -/
+def projectivization_setoid : setoid { v : V // v ≠ 0 } :=
+(mul_action.orbit_rel Kˣ V).comap coe
+
+/-- The projectivization of the `K`-vector space `V`. -/
+def projectivization := quotient (projectivization_setoid K V)
+
+namespace projectivization
+
+variables {V}
+
+/-- Construct an element of the projectivization from a nonzero vector. -/
+def mk (v : V) (hv : v ≠ 0) : projectivization K V := quotient.mk' ⟨v,hv⟩
+
+/-- A variant of `projectivization.mk` in terms of a subtype. `mk` is preferred. -/
+def mk' (v : { v : V // v ≠ 0 }) : projectivization K V := quotient.mk' v
+
+@[simp] lemma mk'_eq_mk (v : { v : V // v ≠ 0}) :
+  mk' K v = mk K v v.2 :=
+by { dsimp [mk, mk'], congr' 1, simp }
+
+noncomputable instance [nontrivial V] : inhabited (projectivization K V) :=
+let e := exists_ne (0 : V) in ⟨mk K e.some e.some_spec⟩
+
+variable {K}
+
+/-- Choose a representative of `v : projectivization K V` in `V`. -/
+protected noncomputable def rep (v : projectivization K V) : V := v.out'.1
+
+lemma rep_nonzero (v : projectivization K V) : v.rep ≠ 0 := v.out'.2
+
+@[simp]
+lemma mk_rep (v : projectivization K V) :
+  mk K v.rep v.rep_nonzero = v :=
+by { dsimp [mk, projectivization.rep], simp }
+
+open finite_dimensional
+
+/-- Consider an element of the projectivization as a submodule of `V`. -/
+protected def submodule (v : projectivization K V) : submodule K V :=
+quotient.lift_on' v (λ v, K ∙ (v : V)) $ begin
+  rintro ⟨a, ha⟩ ⟨b, hb⟩ ⟨x, (rfl : x • b = a)⟩,
+  exact (submodule.span_singleton_smul_eq _ x.ne_zero),
+end
+
+variable (K)
+
+lemma exists_of_mk_eq_mk  (v w : V) (hv : v ≠ 0) (hw : w ≠ 0) (h : mk K v hv = mk K w hw) :
+  ∃ (a : Kˣ), a • w = v :=
+quotient.exact' h
+
+lemma exists_smul_eq_mk_rep
+  (v : V) (hv : v ≠ 0) : ∃ (a : Kˣ), a • v = (mk K v hv).rep :=
+exists_of_mk_eq_mk _ _ _ (rep_nonzero _) hv (by simp)
```

Comment:
A bit longer, but this makes it clear that this is just the standard `mk_out'` lemma
```suggestion
show (projectivization_setoid K V).rel _ _, from quotient.mk_out' ⟨v, hv⟩
```

Reply:
Done